src/Operations.jl

```diff
@@ -1130,10 +1130,14 @@ function free(ctx::Context, pkgs::Vector{PackageSpec})
     end
 end
 
-function gen_test_code(testfile::String; coverage=false)
+function gen_test_code(testfile::String;
+        coverage=false,
+        julia_args::AbstractVector{<:AbstractString}=String[],
+        test_args::AbstractVector{<:AbstractString}=String[])
     code = """
         $(Base.load_path_setup_code(false))
         cd($(repr(dirname(testfile))))
+        append!(empty!(ARGS), $(repr(test_args)))
```

Comment:
Could `repr` here fail for non-`Vector{String}`? Maybe we should just collect the arguments to `Vector{String}` at the entry point.

Reply:
I've implemented your `julia_args = String[a for a in julia_args]`  and `test_args = String[a for a in test_args]` ideas from below, which should solve this.

Reply:
Done.